### PR TITLE
Fixed Trying to remove a view index above child count 0

### PR DESCRIPTION
**Pull request by mattijsf**
Fixes a regressed issue: https://github.com/nitaliano/react-native-mapbox-gl/issues/1579

A PR fixing the issue was initially accepted in PR https://github.com/nitaliano/react-native-mapbox-gl/pull/1274 but parts of it were lost in a recent merge (see above issue)

### CODE REVIEW

**Review by kristfal — APPROVED** [review]
Looks good.

**Comment by jruddell** [comment]
@mattijsf can we merge this? I have a few crashes because of this issue

**Comment by mattijsf** [comment]
@jruddell Sure, but I'm not an official maintainer of this repo so I can't do it.

@kristfal Is there anything preventing this to be merged?

**Comment by sfratini** [comment]
@mattijsf Hey I can merge too. You were the original author of the first PR right? I just want to be sure I am not breaking anything and you know the code already.

**Comment by mattijsf** [comment]
It is not breaking anything. We are using it in our internal fork which is up-to-date with current master. Like described above, these changes where there before but got lost in a big merge(conflict) some time ago :)

**Comment by sfratini** [comment]
There you go :)

**Comment by jruddell** [comment]
thanks guys!

**Pull request merged**
